spark-connector/src/main/java/com/datastrato/gravitino/spark/connector/SparkTransformConverter.java

```diff
@@ -0,0 +1,243 @@
+      Preconditions.checkState(distribution == null, "Should only set distribution once");
+      this.distribution = distributionInfo;
+    }
+
+    private void setSortOrders(SortOrder[] sortOrdersInfo) {
+      Preconditions.checkState(sortOrders == null, "Should only set sort orders once");
+      this.sortOrders = sortOrdersInfo;
+    }
+  }
+
+  public static Transform[] toGravitinoPartitions(
+      org.apache.spark.sql.connector.expressions.Transform[] transforms) {
+    if (ArrayUtils.isEmpty(transforms)) {
+      return Transforms.EMPTY_TRANSFORM;
+    }
+
+    return Arrays.stream(transforms)
+        .filter(
+            transform ->
+                !((transform instanceof BucketTransform)
+                    || (transform instanceof SortedBucketTransform)))
+        .map(
+            transform -> {
+              if (transform instanceof IdentityTransform) {
+                IdentityTransform identityTransform = (IdentityTransform) transform;
+                return Transforms.identity(identityTransform.reference().fieldNames());
+              } else {
+                throw new NotSupportedException(
+                    "Doesn't support Spark transform: " + transform.name());
+              }
+            })
+        .toArray(Transform[]::new);
+  }
+
+  public static DistributionAndSortOrdersInfo toGravitinoDistributionAndSortOrders(
+      org.apache.spark.sql.connector.expressions.Transform[] transforms) {
+    DistributionAndSortOrdersInfo bundles = new DistributionAndSortOrdersInfo();
+    if (ArrayUtils.isEmpty(transforms)) {
+      return bundles;
+    }
+
+    Arrays.stream(transforms)
+        .forEach(
+            transform -> {
+              if (transform instanceof SortedBucketTransform) {
+                Pair<Distribution, SortOrder[]> pair =
+                    toGravitinoDistributionAndSortOrders((SortedBucketTransform) transform);
+                bundles.setDistribution(pair.getLeft());
+                bundles.setSortOrders(pair.getRight());
+              } else if (transform instanceof BucketTransform) {
+                BucketTransform bucketTransform = (BucketTransform) transform;
+                Distribution distribution = toGravitinoDistribution(bucketTransform);
+                bundles.setDistribution(distribution);
+              }
+            });
+
+    return bundles;
+  }
+
+  public static org.apache.spark.sql.connector.expressions.Transform[] toSparkTransform(
+      com.datastrato.gravitino.rel.expressions.transforms.Transform[] partitions,
+      Distribution distribution,
+      SortOrder[] sortOrder) {
+    List<org.apache.spark.sql.connector.expressions.Transform> sparkTransforms = new ArrayList<>();
+    if (ArrayUtils.isNotEmpty(partitions)) {
+      Arrays.stream(partitions)
+          .forEach(
+              transform -> {
+                Preconditions.checkArgument(
+                    transform instanceof SingleFieldPartitioning,
```

Comment:
Here should use `Transform.SingleFieldTransform` instead after #2630 merged

Reply:
ok

Reply:
updated